From 862c36d385917c533a10eba7da2425d1a3e0d8aa Mon Sep 17 00:00:00 2001
From: dianlujitao <dianlujitao@lineageos.org>
Date: Sat, 16 Mar 2019 12:28:11 +0800
Subject: [PATCH 2/2] sepolicy: Allow perf HAL to set mpctl props

 * ro.min_freq_0 and ro.min_freq_4 are renamed to ones with vendor.
   prefix on P, and labeled as vendor_mpctl_prop
 * Addresses the following errors caught in a log:

    [  165.283569] selinux: avc:  denied  { set } for property=vendor.min_freq_0 pid=544 uid=0 gid=0 scontext=u:r:hal_perf_default:s0 tcontext=u:object_r:vendor_mpctl_prop:s0 tclass=property_service permissive=1

This reverts commit 4a25e1746040ddb3a79c7620df1736d00e63f146.

Change-Id: I2667b33de063410b93f6805c8fa66ce6ff028eed
---
 vendor/common/hal_perf_default.te | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/vendor/common/hal_perf_default.te b/vendor/common/hal_perf_default.te
index eb0fbff6..23bd2900 100644
--- a/vendor/common/hal_perf_default.te
+++ b/vendor/common/hal_perf_default.te
@@ -41,6 +41,7 @@ allow hal_perf_client hal_perf_hwservice:hwservice_manager find;
 allow hal_perf cgroup:file r_file_perms;
 allow hal_perf_default proc:file rw_file_perms;
 allow hal_perf device_latency:chr_file rw_file_perms;
+allow hal_perf freq_prop:file r_file_perms;
 allow hal_perf vendor_mpctl_prop:file r_file_perms;
 allow hal_perf_default mpctl_data_file:dir rw_dir_perms;
 allow hal_perf_default mpctl_data_file:file create_file_perms;
@@ -51,8 +52,8 @@ hal_client_domain(hal_perf_default, hal_graphics_composer);
 allow hal_perf_default qdisplay_service:service_manager find;
 r_dir_file(hal_perf_default, appdomain);
 
-# Allow perf HAL to set freq props
-set_prop(hal_perf_default, freq_prop)
+# Allow perf HAL to set mpctl props
+set_prop(hal_perf_default, vendor_mpctl_prop)
 
 allow hal_perf {
     sysfs_devices_system_cpu
-- 
2.17.1

